R/GitHub.R

```diff
@@ -0,0 +1,374 @@
+#' @importFrom R6 R6Class
+#' @importFrom dplyr mutate
+#' @importFrom magrittr %>%
+#' @importFrom rlang %||%
+#'
+#' @title A GitHub API Client class
+#' @description An object with methods to obtain information form GitHub API.
+
+GitHub <- R6::R6Class("GitHub",
+  inherit = GitService,
+  cloneable = FALSE,
+  public = list(
+
+    #' @description  A method to list all repositories for an organization,
+    #'   a team or by a keyword.
+    #' @param orgs A character vector of organizations (owners of repositories).
+    #' @param by A character, to choose between: \itemize{\item{org - organizations
+    #'   (owners of repositories)} \item{team - A team} \item{phrase - A keyword in
+    #'   code blobs.}}
+    #' @param team A list of team members. Specified by \code{set_team()} method
+    #'   of GitStats class object.
+    #' @param phrase A character to look for in code blobs. Obligatory if
+    #'   \code{by} parameter set to \code{"phrase"}.
+    #' @param language A character specifying language used in repositories.
+    #' @return A data.frame of repositories.
+    get_repos = function(orgs = self$orgs,
```

Comment:
You mean you would just call `self$orgs` inside the function instead of putting it as argument?

Reply:
I tried to go with the functional paradigm spirit (not sure now if I catch it correctly) where it is good to define everything that is going on in the function. On the other hand it saves some space/time when writing e.g. `api_url` in the body instead of `self$api_url`.

Reply:
I removed all superfluous parameters like `token` and `api_url` in https://github.com/r-world-devs/GitStats/pull/89